From deb57e06d3edc0c5784b956e2c31c82b04b171a1 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 30 Jan 2025 11:11:00 +0100
Subject: [PATCH 1500/1695] dts: st: move node hash1 and rng1 in
 stm32mp211.dtsi

HASH1 and RNG1 peripheral are available on stm32mp21xa and xd SoCs.
Move node hash1 and rng1 from stm32mp21xc.dtsi to stm32mp211.dtsi

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 42a1fe42e2d2527cd79472b4eca37b81c938d814 dts: st: move node hash1 and rng1 in stm32mp211.dtsi]
Change-Id: I10825aa1095842c0b4fd07f56058fd6da569a053
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433032
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi  | 18 ++++++++++++++++++
 core/arch/arm/dts/stm32mp21xc.dtsi | 18 ------------------
 2 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index dc8c246d1..8e7114014 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -426,6 +426,24 @@
 				resets = <&rcc RNG2_R>;
 			};
 
+			hash1: hash@42030400 {
+				compatible = "st,stm32mp13-hash";
+				reg = <0x42030400  0x400>;
+				interrupts = <GIC_SPI 132 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_HASH1>;
+				resets = <&rcc HASH1_R>;
+				status = "disabled";
+			};
+
+			rng1: rng@42030800 {
+				compatible = "st,stm32mp21-rng";
+				reg = <0x42030800 0x400>;
+				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG1>;
+				clock-names = "rng_clk", "rng_hclk";
+				resets = <&rcc RNG1_R>;
+				status = "disabled";
+			};
+
 			iwdg1: watchdog@44010000 {
 				compatible = "st,stm32mp1-iwdg";
 				reg = <0x44010000 0x400>;
diff --git a/core/arch/arm/dts/stm32mp21xc.dtsi b/core/arch/arm/dts/stm32mp21xc.dtsi
index 17bbddf13..fe09c5e5a 100644
--- a/core/arch/arm/dts/stm32mp21xc.dtsi
+++ b/core/arch/arm/dts/stm32mp21xc.dtsi
@@ -14,24 +14,6 @@
 		status = "disabled";
 	};
 
-	hash1: hash@42030400 {
-		compatible = "st,stm32mp13-hash";
-		reg = <0x42030400  0x400>;
-		interrupts = <GIC_SPI 132 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&rcc CK_BUS_HASH1>;
-		resets = <&rcc HASH1_R>;
-		status = "disabled";
-	};
-
-	rng1: rng@42030800 {
-		compatible = "st,stm32mp21-rng";
-		reg = <0x42030800 0x400>;
-		clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG1>;
-		clock-names = "rng_clk", "rng_hclk";
-		resets = <&rcc RNG1_R>;
-		status = "disabled";
-	};
-
 	saes: saes@42030c00 {
 		compatible = "st,stm32mp13-saes";
 		reg = <0x42030c00 0x400>;
-- 
2.47.2

